eldritch/gate_encounters.py

```diff
@@ -20,64 +23,393 @@ def encounter_event(self, character, world_name):
+
+
+def Dreamlands8(char) -> events.Event:
+  check = events.Check(char, "lore", 0)
+  freeze = events.LostInTimeAndSpace(char)
+  return events.PassFail(char, check, events.Return(char, char.place.info.name), freeze)
+
+
+def Pluto8(char) -> events.Event:
+  check = events.Check(char, "speed", -1)
+  statue = events.Gain(char, {"dollars": 5, "clues": 2})
+  lost = events.LostInTimeAndSpace(char)
+  return events.PassFail(char, check, statue, lost)
+
+
+def Other8(char) -> events.Event:
+  return events.Draw(char, "common", 1)
+
+
+def Dreamlands9(char) -> events.Event:
+  check = events.Check(char, "sneak", 0)
+  lose_item_count = values.Calculation(
+      values.ItemDeckCount(char, {"common", "unique", "spells", "tradables"}), None,
+      values.ceildiv, 2,
+  )
+  lose_money_count = values.Calculation(char, "dollars", values.ceildiv, 2)
+  loss = events.Sequence([
+      events.LoseItems(char, lose_item_count),
+      events.Loss(char, {"dollars": lose_money_count})
+  ], char)
+  return events.PassFail(char, check, events.Nothing(), loss)
+
+
+def Pluto9(char) -> events.Event:
+  check = events.Check(char, "will", -2)
+  loss = events.LoseItems(char, 1)
+  return events.PassFail(char, check, events.Nothing(), loss)
+
+
+def Other9(char) -> events.Event:
+  return events.Loss(char, {"sanity": 1})
+
+
+def Dreamlands10(char) -> events.Event:
   check = events.Check(char, "luck", -1)
   loss = events.Loss(char, {"stamina": 1})
   delay = events.Delayed(char)
   return events.PassFail(char, check, events.Nothing(), events.Sequence([loss, delay], char))
 
 
-def Abyss10(char):
+def Abyss10(char) -> events.Event:
   check = events.Check(char, "luck", -2)
   loss = events.Loss(char, {"sanity": 1})
   delay = events.Delayed(char)
   return events.PassFail(char, check, events.Nothing(), events.Sequence([loss, delay], char))
 
 
-def Other10(char):
+def Other10(char) -> events.Event:
   check = events.Check(char, "lore", -1)
   delay = events.Delayed(char)
   return events.PassFail(char, check, events.Nothing(), delay)
 
 
-def Plateau16(char):
+def Pluto11(char) -> events.Event:
+  return events.MonsterAppears(char)
+
+
+def GreatHall11(char) -> events.Event:
+  check = events.Check(char, "luck", -1)
+  dice_roll = events.DiceRoll(char, 1)
+  prison = events.PassFail(char, dice_roll, events.Delayed(char), events.MonsterAppears(char))
+  return events.PassFail(char, check, events.Nothing(), prison)
+
+
+def Other11(char) -> events.Event:
+  check = events.Check(char, "luck", -1)
+  weapon = events.Draw(char, "common", 1, target_type=items.Weapon)
+  return events.PassFail(char, check, weapon, events.Nothing())
+
+
+def Pluto12(char) -> events.Event:
+  check = events.Check(char, "will", -2)
+  san_loss = events.Loss(char, {"sanity": 2})
+  spell_loss = events.LoseItems(char, 2, "Lose spells", {"spells"})
+  prereq = values.ItemDeckPrerequisite(char, "spells", threshold=2)
+  choice = events.BinaryChoice(
```

Comment:
Instead of creating a second pass/fail, you can pass the prereq object directly into BinaryChoice (it will get attached to the ChoiceEvent, and the choice will be greyed out if the user does not meet the prerequisite)